### PR TITLE
[netcore] Implement Object.GetRawData

### PR DESCRIPTION
**Pull request by filipnavara**

### CODE REVIEW

**Comment by filipnavara** [comment]
I forgot there is intrinsic for it so it never gets executed but I think we can safely use the managed code instead.

**Comment by vargaz** [comment]
Note that these kinds of code only have the same perf if you have an optimizing compiler like llvm or the coreclr jit, it will be slower with the mono JIT or interpreter. So we should keep the intrinsics.

**Comment by filipnavara** [comment]
I checked the codegen and in both cases it ends up `long_add_imm %rax <- %r15 [16]` but the managed code generated an explicit null check for some reason.

**Comment by filipnavara** [comment]
Ok, the NULL check comes from here: https://github.com/mono/mono/blob/9cc5ea3862894a10d87a51e302893b0165e4850c/mono/mini/method-to-ir.c#L9265-L9268

**Comment by filipnavara** [comment]
The null check actually gets eliminated when the code gets inlined on the sites so it should work even in non-LLVM case.

File: netcore/System.Private.CoreLib/src/System/Object.cs [hunk 1]
```diff
@@ -15,7 +16,7 @@ partial class Object
 		protected extern object MemberwiseClone ();
 
 		[Intrinsic]
-		internal ref byte GetRawData () => throw new NotImplementedException ();
+		internal ref byte GetRawData () => ref Unsafe.As<Mono.RawData>(this).Data;
```
Comment (marek-safar) [inline netcore/System.Private.CoreLib/src/System/Object.cs]: Not in favour of this. We have Intrinsic why to have managed code and dependencies which will never be used?

Comment (filipnavara) [reply netcore/System.Private.CoreLib/src/System/Object.cs]: The idea was to remove the intrinsic in favour of the managed code. It seems to produce the same codegen.

Comment (marek-safar) [reply netcore/System.Private.CoreLib/src/System/Object.cs]: I don't think it's worth it, as we already have the instrincis and the managed code is not really improving the readability or performance.

Comment (filipnavara) [reply netcore/System.Private.CoreLib/src/System/Object.cs]: I beg to differ on the readability but I am fine with closing this as the improvement is questionable.

Comment (filipnavara) [reply netcore/System.Private.CoreLib/src/System/Object.cs]: Also, we can shared the implementation with CoreCLR which uses basically the same code as I used here.

Comment (jaykrell) [reply netcore/System.Private.CoreLib/src/System/Object.cs]: Is the interpreter keeping up with the intrinsics?

Comment (filipnavara) [reply netcore/System.Private.CoreLib/src/System/Object.cs]: For this particular case the interpreter has both `object.GetRawData` (netcore only) and `Unsafe.As`.

Comment (jaykrell) [reply netcore/System.Private.CoreLib/src/System/Object.cs]: I don't mean performance, but in recognition.
Having the IL loosens the requirement of keeping up with recognition.

**Pull request closed without merging**